### ISSUE TITLE
Intel AMT certificate provisioned with wrong common name FQDN | only works by IP

### ISSUE THREAD

**Issue by k1n6b0b** [Ylianst/MeshCentral#6592]
**Describe the bug**
The certificate loaded during provisioning does not use the dedicated FQDN specified in AMT config, im not sure how to specify it (or reprovision?)

**To Reproduce**

Configure AMT to use a dedicated IP and dedicated FQDN
Provision with MeshCentral

`curl https://10.0.0.124:16993 --insecure -v #works `

`curl https://IntelAMT-6398b0b2a9ff:16993 --insecure -v --resolve IntelAMT-6398b0b2a9ff:16993:10.0.0.124 #works`

However 
`curl https://dedicated-fqdn.xxx.yyy --insecure -v `
fails because the certificate doesnt match

Certificate:
<img width="535" alt="image" src="https://github.com/user-attachments/assets/43ea2e77-5c6e-4b0a-9140-b9969522f7ce">

AMT config
<img width="658" alt="image" src="https://github.com/user-attachments/assets/5054f9b1-fc76-4998-9c49-d786c3df7500">

**Expected behavior**
Maybe i did something out of order, but i'd expect the certificate loaded by MeshCentral to match the Computer Host Name . Domain Name

**Screenshots**
See above

**Server Software (please complete the following information):**
 - OS: Debian 12
 - Virtualization: Proxmox LXC
 - Network: Local certificate by certbot (symlink'd)
 - Version: v1.1.33
 - Node: v20.18.1

**Client Device (please complete the following information):**
 - Device: Intel BNUC11TNHv70L00
 - OS: Proxmox
 - Network: Local
 - Browser: Google Chrome

**Additional context**

**Your config.json file**
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove th>
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "meshcentral.xxx.yyy",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "_redirAliasPort": 80,
    "_port": 80,
    "_redirPort": 81,
    "_aliasPort": 443,
    "_TLSOffload": false
  },
  "domains": {
    "": {
      "title": "XXX",
      "title2": "meshcentral",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true,
      "agentConfig": [
        "webSocketMaskOverride=1"
      ],
      "certUrl": "https://meshcentral.xxx.yyy:443/"
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "skipChallengeVerification": true,
    "production": false
  }
}

```

**Comment by si458** [comment]
this is not a bug,
the AMT on the remote device will have its own self-signed intel AMT certificate `https://127.0.0.1:16993`
and the same goes for MeshCentral service side
it will create a self-signed certificate for the AMT `https://meshcentral.myserver.com:4433`

**Comment by k1n6b0b** [comment]
So the issue I'm having is none of my AMT CIRA stuff is working, maybe i'm following the wrong thread to debug
```

AMT: Start Management node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 2
AMT: pve4 Checking Intel AMT state...
AMT: pve4 Attempt Initial Contact CIRA-LMS
AMT: pve4 LMS-Connect TLS admin
AMT: pve4 Initial Contact Response 408
AMT: pve4 Attempt Initial Contact CIRA-LMS
AMT: pve4 LMS-Connect NoTLS admin
AMT: pve4 Initial Contact Response 408
AMT: pve4 Remove device node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 2 14

```
<img width="503" alt="image" src="https://github.com/user-attachments/assets/c6273f41-5540-4f10-9432-830fa453ab9a">

<img width="907" alt="image" src="https://github.com/user-attachments/assets/89207dd4-b904-49c4-ae8d-4179341ecf08">

**Comment by si458** [comment]
you are using linux (guessing from the pve4 hostname)
intel amt and linux doesnt seem to be working correctly, the is QUITE A LOT of open issues about it
im going to try have a look next week with an v12 amt device i have and find out whats what
im the mean time, how did u setup the amt on the device to begin with?
setup a meshagent group, set amt policy to fully automatic, instal meshagent, use amtconfig in console?

**Comment by k1n6b0b** [comment]
So to get it out of CCM --

Rebooted host, Control-P
Changed default password 'admin' to something new
completely disabled AMT
Rebooted, Control-P
enabled AMT
Set user policy to Never
provisioned network to DHCP (didnt work)
set DNS per above

Saved/Rebooted

Now i was able to see and edit settings with meshcmd, but I was unable to see any DHCP address

I was able to ssh access https://localhost:16993 and login, here I changed DHCP -> Static IP
Rebooted the host

Now I was able to access remotely to the Static IP i set (which brings me to the above post, checking the cert etc) 

**Note:** I do have 2 NICs in a bridge0 in proxmox -- previously i've seen issues with IPMI and VMware nic-teaming on other hardware, so noting it here
**Note:** this one also seemed possibly related: https://github.com/Ylianst/MeshCentral/issues/6281

**Comment by si458** [comment]
ok so
1. Fully Automatic will try to activate ur machine into ACM mode, and if it fails it falls back to CCM, then if that fails, it just gives up - THIS IS EXPECTED!
2. ALWAYS USE DHCP mode in the network settings, NEVER USE STATIC IP AS THIS BREAKS CIRA!!! VERY IMPORTANT!
3. when u setup amt to begin with, i would set the dns suffix to your meshcentral dns name

**Comment by k1n6b0b** [comment]
1) - Right, since i kept getting CCM I was trying to force ACM, makes sense
2) Interesting.. I'm unable to get a DHCP lease -- so this is a sticking point, it only worked when set statically
3) DNS suffix matches, as in xxx.yyy and meshcentral.xxx.yyy -- is that correct, or did you mean the dns suffix to be the FQDN of meshcentral?

**Comment by si458** [comment]
yes unfortunately #6281 had the same issue they only have static ip ranges,
but unfortunately the `Environment Detection` disables CIRA work working if you set a STATIC IP in the AMT
this is an AMT bug and chances are will NEVER be fixed unless u bugged Intel to fix it

as for the dns suffix inside the amt network settings, it must match with whatevever your meshcentral server uses i have found
so for me i use `meshcentral.mydomain.com` and matches my `cert` value in config.json

**Comment by k1n6b0b** [comment]
OK I'll play around more with DHCP then. just tried it and it loses its address to 0.0.0.0

```
> amt
{
  core-ver: 1
  OsHostname: "pve4"
  Flags: 4
  Versions: {
    Flash: "15.0.35"
    Netstack: "15.0.35"
    AMTApps: "15.0.35"
    AMT: "15.0.35"
    Sku: "16392"
    VendorID: "8086"
    Build Number: "1951"
    Recovery Version: "15.0.35"
    Recovery Build Num: "1951"
    Legacy Mode: "False"
  }
  UUID: "2a1b708e-6c0a-dd84-7bcb-48210b37300b"
  ProvisioningMode: 1
  ProvisioningState: 2
  net0: {
    enabled: 1
    dhcpEnabled: 1
    dhcpMode: "ACTIVE"
    mac: "48:21:0B:37:30:0B"
    address: "0.0.0.0"
  }
}
```

**Comment by si458** [comment]
@k1n6b0b thats correct, intel amt in dhcp mode never knows what the IP address of the device is
dont ask why, so it will always show 0.0.0.0

**Comment by k1n6b0b** [comment]
Should I see a lease in my DHCP server tho? (I do not see one)

**Comment by si458** [comment]
> Should I see a lease in my DHCP server tho? (I do not see one)

from what i recall, yes u should do when the computer is switched off,
but if u have a bond interface, the dhcp might not work because ur switch is sending the dhcp request over the bonded interfaces and when the pc is OFF, the bond isnt active

**Comment by k1n6b0b** [comment]
Interesting, at some point it pulled a lease, i see it in my DHCP server, but its inaccessible (maybe it was for a bit) 

```
AMT: Start Management node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 0
AMT: pve4 Checking Intel AMT state...
AMT: pve4 Attempt Initial Contact CIRA
AMT: pve4 CIRA-Connect NoTLS admin XXXXXX
AMT: pve4 Stop Management node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 0
AMT: pve4 Remove device node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 0 1
AMT: Start Management node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 0
AMT: pve4 Checking Intel AMT state...
AMT: pve4 Attempt Initial Contact CIRA
AMT: pve4 CIRA-Connect NoTLS admin XXXXXX
AMT: pve4 Initial Contact Response 200
AMT: pve4 Intel AMT connected.
AMT: pve4 Done.
```

maybe its unstable... as its not accessible by ping, curl, or meshcentral tools now

```
AMT: pve4 Stop Management node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 0
AMT: pve4 Remove device node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 0 1
AMT: Start Management node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 0
AMT: pve4 Checking Intel AMT state...
AMT: pve4 Attempt Initial Contact CIRA
AMT: pve4 CIRA-Connect NoTLS admin XXXXX
AMT: Start Management node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 2
AMT: pve4 Checking Intel AMT state...
AMT: pve4 Attempt Initial Contact CIRA-LMS
AMT: pve4 LMS-Connect TLS admin
AMT: pve4 Initial Contact Response 408
AMT: pve4 Attempt Initial Contact CIRA-LMS
AMT: pve4 LMS-Connect NoTLS admin
AMT: pve4 Initial Contact Response 408
AMT: pve4 Remove device node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 2 14
AMT: pve4 Stop Management node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 0
AMT: pve4 Remove device node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 0 1
AMT: Start Management node//IsmcdV9$fjIO4@B4IHO9jhrTb6IfxGR8TGB61kTE5mH@I@46sGmIQ58Hj6Pflox8 0
AMT: pve4 Checking Intel AMT state...
AMT: pve4 Attempt Initial Contact CIRA
AMT: pve4 CIRA-Connect NoTLS admin XXXXX
```

**Comment by k1n6b0b** [comment]
rebooted the host again and its working!!

![image](https://github.com/user-attachments/assets/ee838e25-d7a4-46e0-b4f5-9d691136a687)

cannot connect to the IP leased by my DHCP server, or localhost.. i have no idea what IP its using rn but its working.

Stability def seems to be an issue, but sounds like per your note that the drive is wonky AF. I'll keep playing with it.

Let me know if i can provide anything more to help

**Comment by k1n6b0b** [comment]
I was also trying this, but unclear if it made any difference. 

https://community.intel.com/t5/Intel-vPro-Platform/AMT-stops-responding-via-network-DHCP-IP-Address-missing/td-p/1240125/page/3

On reboot, the _Current Power Mode_ is always **auto** until I change it to **on**

**Comment by si458** [comment]
@k1n6b0b I'm having a similar issue at the moment with my AMT v12 machine, it seems to be flaky with its DHCP lease and gets it when it feels like it as i can't ping it and get a reply but that's for the info about power mode, I will do some investigating

**Comment by marek26340** [comment]
> AMT: pve4 LMS-Connect TLS admin
> AMT: pve4 Initial Contact Response 408
> AMT: pve4 Attempt Initial Contact CIRA-LMS
> AMT: pve4 LMS-Connect NoTLS admin

Sounds very similar to my issue. #6565 
Still haven't figured this out. I think I'll just try recompiling OpenSSL so that it'll be clear of any modifications made by Canonical/Ubuntu regarding legacy TLS 1.0/1.1 support. I will report back in my own issue if I'll have any news...
